src/main/java/com/routeme/api/UserController.java

```diff
@@ -34,61 +34,63 @@ public UserController(UserService service) {
 
     @RequestMapping(method = RequestMethod.GET, produces = "application/json")
     @ResponseStatus(HttpStatus.OK)
-    List<UserDTO> findAll() {
+    private List<UserDTO> findAll() {
         return service.findAll();
     }
 
     @RequestMapping(method = RequestMethod.POST, consumes = "application/json", produces = "application/json")
     @ResponseStatus(HttpStatus.CREATED)
-    UserDTO create(@RequestBody @Valid UserDTO userEntry) {
-        UserDTO userRecord = service.create(userEntry);
-        String userId = userRecord.getEmail();
-        predictionIOClient.addUserToClient(userId);
-        // SHOULD BE BATCHED
-        predictionIOClient.preferRouteType(userId, PredictionIOClient.PREFERENCE_ROUTE_TYPE_LEASTTIME);
-        predictionIOClient.preferRouteMode(userId, PredictionIOClient.PREFERENCE_ROUTE_MODE_BUS);
-        predictionIOClient.preferRouteMode(userId, PredictionIOClient.PREFERENCE_ROUTE_MODE_BUS);
-        predictionIOClient.preferRouteMode(userId, PredictionIOClient.PREFERENCE_ROUTE_MODE_UBAHN);
+    private UserDTO signup(@RequestBody @Valid UserDTO userEntry) {
+        UserDTO userRecord = service.create(userEntry).orElseThrow(
+                () -> new UserAlreadyExistsException(userEntry.getEmail()));
+        return userRecord;
+    }
+
+    @RequestMapping(value = "/login", method = RequestMethod.POST, consumes = "application/json", produces = "application/json")
+    @ResponseStatus(HttpStatus.FOUND)
+    private UserDTO login(@RequestBody UserDTO userEntry) {
+        UserDTO userRecord = service.findByCredentials(userEntry.getEmail(), userEntry.getPassword()).orElseThrow(
+                () -> new WrongCredentialsException());
         return userRecord;
     }
 
     @RequestMapping(value = "/take", method = RequestMethod.POST, produces = "application/json")
     @ResponseStatus(HttpStatus.CREATED)
-    void takeRoute(@RequestBody @Valid EventDTO eventEntry) {
+    private void takeRoute(@RequestBody @Valid EventDTO eventEntry) {
         predictionIOClient.takeRoute(eventEntry.getUserId(), eventEntry.getTargetEntityId());
     }
 
     @RequestMapping(value = "/view", method = RequestMethod.POST, produces = "application/json")
     @ResponseStatus(HttpStatus.CREATED)
-    void viewRoute(@RequestBody @Valid EventDTO eventEntry) {
+    private void viewRoute(@RequestBody @Valid EventDTO eventEntry) {
         predictionIOClient.viewRoute(eventEntry.getUserId(), eventEntry.getTargetEntityId());
     }
 
     @RequestMapping(value = "/viewlast", method = RequestMethod.POST, produces = "application/json")
     @ResponseStatus(HttpStatus.CREATED)
-    void ViewRouteLast(@RequestBody @Valid EventDTO eventEntry) {
+    private void ViewRouteLast(@RequestBody @Valid EventDTO eventEntry) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method names should not start with capital letters](https://www.codacy.com/app/heshamhamdymassoud/RouteMe-API/file/2619992100/issues/source?bid=3287636&fileBranchId=3309525#l71)